libs/ipc/inc/public/tfc/ipc/packet.hpp

```diff
@@ -148,9 +151,13 @@ struct packet {
         std::copy_n(buffer_iter, result.header.value_size - 1, reinterpret_cast<std::byte*>(&substitute));
         result.value = substitute * value_t::value_type::reference;
       } else {
-        typename value_t::error_type substitute{};
-        std::copy_n(buffer_iter, result.header.value_size - 1, reinterpret_cast<std::byte*>(&substitute));
-        result.value = std::unexpected{ substitute };
+        if constexpr (std::is_same_v<typename value_t::error_type, mass_error_e>) {
+          result.value = std::unexpected{ static_cast<mass_error_e>(*buffer_iter) };
+        } else {
+          typename value_t::error_type substitute{};
+          std::copy_n(buffer_iter, result.header.value_size - 1, reinterpret_cast<std::byte*>(&substitute));
+          result.value = std::unexpected{ substitute };
+        }
```

Comment:
I don't like this being explicitly done for this error enum, how is this broken like it is now?

Reply:
https://github.com/Skaginn3x/framework/actions/runs/7248918924/job/19745806361

this is the error that happens, 

```bash
 forming offset 1 is out of the bounds [0, 1] of object 'substitute' with type 'std::expected<mp_units::quantity<mp_units::si::milli_<mp_units::si::gram()>(), long int>, tfc::ipc::details::mass_error_e>::error_type' {aka 'tfc::ipc::details::mass_error_e'}
```

Reply:
Is this only an error for gcc release mode?
Is this a warning or an actual error?

Reply:
This is only an error for gcc-release, it is an error, not warning.